.github/workflows/ci.yml

```diff
@@ -327,6 +325,26 @@ jobs:
           python make.py python-ci-install
         timeout-minutes: 20
 
+      # We only save the libparsec lib when:
+      # - We are not in a github queue branch (they're a one time use so caching won't help)
+      # - The rust code isn't modifed.
+      # - We haven't add a cache it.
+      - name: Save cached libparsec to be reused on later call
+        if: >-
+          steps.python-changes.outputs.run == 'true'
```

Comment:
Can't we just use `steps.cache-libparsec.outputs.cache-hit != 'true' && contains(github.ref, 'gh-readonly-queue') != 'true'` ?

i.e. we want to save the cache if it wasn't already present and if we are not in a github queue branch

(the idea is to let `cache-libparsec` job determine whether or not libparsec should be built, then not duplicate this check here to avoid inconsistency bugs)

Reply:
Almost perfect :p

Just missing the part for the PR request ;)

```txt
steps.cache-libparsec.outputs.cache-hit != 'true' # We have a cache hit, so we don't need to save again
&& contains(github.ref, 'gh-readonly-queue') != 'true' # We are in a queue branch so we don't save it
&& (contains(github.ref, 'refs/pull') == 'true' && steps.changes.outputs.rust-changes == 'true') != 'true' # We are in a PR and the rust code have changed
```

Reply:
WDYT ?

Reply:
we indeed want to keep in cache the modified libparsec in the branch (because it makes working on the branch much more convenient, and it libparsec is only ~10mo so it's very light on the cache)

> Should we have a condition on the PR ?

I don't understand what you mean by "condition on the PR" is this a github action specific term ?

My point is only:
- if we had a cache hit there is no point in saving the cache
- if we didn't have a cache hit it is either because 1) we didn't try to setup the cache (i.e. because we don't need to test the python code) or 2) because of actual cache miss (i.e. the rust code has changed)

I forgot about cache miss option 1) in my previous comment.

So thing would be more like:
```
# Something has changed among Python code, libparsec Python bindings or libparsec Rust codeSomething needed to generate the Python wheel has changed...
steps.changes.need-run-python-tests == 'true'
# ...and we had to rebuild libparsec
&& steps.cache-libparsec.outputs.cache-hit != 'true'
# ...and we are not in the merge queue branch (where caching is useless given the branch is short lived)
&& contains(github.ref, 'gh-readonly-queue') != 'true'
```

Reply:
:+1:

Reply:
done